rest-api-spec/src/main/resources/rest-api-spec/test/search/90_search_after.yml

```diff
@@ -94,3 +91,126 @@ setup:
 
   - match: {hits.total: 3}
   - length: {hits.hits: 0 }
+
+---
+"date":
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              timestamp:
+                type: date
+                format: yyyy-MM-dd HH:mm:ss.SSS
+  - do:
+      bulk:
+        refresh: true
+        index: test
+        body: |
+          {"index":{}}
+          {"timestamp":"2019-10-21 00:30:04.828"}
+          {"index":{}}
+          {"timestamp":"2019-10-21 08:30:04.828"}
+
+  - do:
+      search:
+        index: test
+        body:
+          size: 1
+          sort: [{ timestamp: desc }]
+  - match: {hits.total.value: 2 }
+  - length: {hits.hits: 1 }
+  - match: {hits.hits.0._index: test }
+  - match: {hits.hits.0._source.timestamp: "2019-10-21 08:30:04.828" }
+  - match: {hits.hits.0.sort: [1571646604828] }
+
+  # search_after with the sort
+  - do:
+      search:
+        index: test
+        body:
+          size: 1
+          sort: [{ timestamp: desc }]
+          search_after: [1571646604828]
+  - match: {hits.total.value: 2 }
+  - length: {hits.hits: 1 }
+  - match: {hits.hits.0._index: test }
+  - match: {hits.hits.0._source.timestamp: "2019-10-21 00:30:04.828" }
+  - match: {hits.hits.0.sort: [1571617804828] }
+
+  # search_after with the formatted date
+  - do:
+      search:
+        index: test
+        body:
+          size: 1
+          sort: [{ timestamp: desc }]
+          search_after: ["2019-10-21 08:30:04.828"]
+  - match: {hits.total.value: 2 }
+  - length: {hits.hits: 1 }
+  - match: {hits.hits.0._index: test }
+  - match: {hits.hits.0._source.timestamp: "2019-10-21 00:30:04.828" }
+  - match: {hits.hits.0.sort: [1571617804828] }
+
+---
+"date_nanos":
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              timestamp:
+                type: date_nanos
+                format: yyyy-MM-dd HH:mm:ss.SSSSSS
+  - do:
+      bulk:
+        refresh: true
+        index: test
+        body: |
+          {"index":{}}
+          {"timestamp":"2019-10-21 00:30:04.828740"}
+          {"index":{}}
+          {"timestamp":"2019-10-21 08:30:04.828733"}
+
+  - do:
+      search:
+        index: test
+        body:
+          size: 1
+          sort: [{ timestamp: desc }]
+  - match: {hits.total.value: 2 }
+  - length: {hits.hits: 1 }
+  - match: {hits.hits.0._index: test }
+  - match: {hits.hits.0._source.timestamp: "2019-10-21 08:30:04.828733" }
+  - match: {hits.hits.0.sort: [1571646604828733000] }
```

Comment:
I wonder if we should use the formatted version in the `sort` output ? We don't allow this format (nanoseconds since epoch) when ingesting or querying a date_nanos field so that's an exception here.

Reply:
I'm honestly not sure! I kind of figured the contact of `search_after` was "take the last sort value and stick it into `search_after`" but I see that we support formatted.

I also wonder what counts as breaking here. If we were to change the result to be formatted I imagine we'll break someone.

Reply:
We do support formatted output so my point was that the `sort` section should return the formatted version in  order to avoid returning long that represents nanoseconds since epoch. I agree that this is not in the scope of this change so let's merge this one and we can discuss the formatting of sort values in a follow up (I can open the issue) ?

Reply:
> I agree that this is not in the scope of this change so let's merge this one and we can discuss the formatting of sort values in a follow up (I can open the issue) ?

:+1: